### ISSUE TITLE
Waterlogged blocks can't hydrate farmland

### ISSUE THREAD

**Issue by wed15** [jbredwards/Fluidlogged-API#76]
**Describe the bug**
Waterlogged blocks don't hydrate farmland ~~or allow sugarcanes to be placed on neighboring sand~~. First realized this in my existing modded survival world (with 290+ mods) after updating Fluidloggedapi v1.6.2 from v1.7.2b. Also reproduceable in new world.

Since this bug doesn't occur in new instance with only forge and this mod installed, it seems like mod conflict, but I'm not sure which mod should be suspected. I'll try to find it if I got time for it later. Any hint for suspected type of mod would be helpful.

**Version used**
Forge: 14.23.5.2855
FluidloggedAPI: 1.7.2b

**Comment by jbredwards** [comment]
My initial guess is that some mod is overriding the vanilla behavior with its own, can you send your log?

**Comment by wed15** [comment]
Here's the log: [latest.log](https://github.com/jbredwards/Fluidlogged-API/files/8382273/latest.log)

Also clarification about sugarcanes - sand (and dirt) actually works fine with fluidlogged blocks, it was snad from Snad mod not working. By the way snad recognizes other modded blocks considered as water (for example fish tank from Just a few fish mod).

and more details about the bug I found later:
- other modded plants which behave like sugarcanes such as cattails from Biomeoplenty, can't be placed on dirt with fluidlogged block
- other modded farmland blocks can't be hydrated as well

**Comment by wed15** [comment]
Found the mod that causing problems, it was [Actually Unbreaking](https://www.curseforge.com/minecraft/mc-mods/actually-unbreaking-forge) mod. Reproduceable with only FluidloggedAPI and this mod. I have no idea how these two mods could conflict.

**Comment by Rongmario** [comment]
That is... quite bizarre, can you send a log file with just these two mods combined in an instance?

**Comment by wed15** [comment]
Here: [latest.log](https://github.com/jbredwards/Fluidlogged-API/files/8384875/latest.log)

**Comment by Rongmario** [comment]
@jbredwards ?
```
[08:54:54] [main/WARN] [mixin]: Reference map 'mixins.mixincompat.refmap.json' for mixins.mixincompat.json could not be read. If this is a development environment you can ignore this message
[08:54:54] [main/WARN] [mixin]: Failed to select mixin config: mixins.fluidlogged_api.json
java.lang.NullPointerException: null
```

**Comment by jbredwards** [comment]
still persists even after fixing the NPE. This issue also effects more than just farmland and seems to prevent all of this mod's mixins from transforming anything, though doesn't interfere with interfaces (ie. IFluidloggable). Also only occurs outside a dev environment, as I couldn't replicate while in one, which is odd...

**Comment by embeddedt** [comment]
@jbredwards Are you using MixinBooter to load the mixins or the original approach of calling `MixinBootstrap.init()` yourself? Maybe the former would help.

**Comment by jbredwards** [comment]
They're being loaded through MixinBooter's `IEarlyMixinLoader` in `ASMHandler`, but that doesn't seem to be helping with the issue

**Comment by wed15** [comment]
While testing alternatives for actually unbreaking mod, I realized farmland and snad problems still remaining (some fluidlogging behaviors are seems back though). This time it was caused by patchFarmland option from Foamfix. Disabling it in config fixes the farmland issue. I didn't had this problem when using FluidloggedAPI v1.6.2.

logs generated with only FluidloggedAPI and Foamfix 0.10.14:
[debug.log](https://github.com/jbredwards/Fluidlogged-API/files/8404824/debug.log)
[latest.log](https://github.com/jbredwards/Fluidlogged-API/files/8404825/latest.log)

**Comment by embeddedt** [comment]
There are a number of outstanding issues in the newer version of FoamFix and I'm not sure they'll be resolved. I would suggest sticking with the older 0.10.11.

**Comment by jbredwards** [comment]
The behaviors in v1.7.x that aren't handled through mixins are unaffected by this issue, so things like rendering & placement, those are handled through asm. I may revert everything back to asm if I can't work out all these mixin-related issues, asm seems far better & more stable imo
> I didn't had this problem when using FluidloggedAPI v1.6.2.

This is because v1.6.2 didn't use any mixins.

**Comment by Rongmario** [comment]
I'd prefer not using mixins when it comes to frequent call sites. But if you do stick with mixins, I'm not sure if you already do this but you can use flag `-Dmixin.debug.export=true` in VM Options to see how the classes look after they've been mixin'd into.

**Comment by jbredwards** [comment]
I've decided that the next version of the mod will not use mixins, marking this as "upcoming fix" as a result.

**Comment by jbredwards** [comment]
fixed in v1.8.0
